### ISSUE TITLE
Blueprint API Populate Where not working

### ISSUE THREAD

**Issue by tbitowt** [balderdashy/sails#3072]
Hi

I have created sample project and created models like:

```
Photo.js:

module.exports = {

  attributes: {
    owner:{
      model:"User"
    }
  }
};
```

```
User.js

module.exports = {

  attributes: {
    photos:{
      collection:"Photo",
      via:"owner"
    }
  }
};
```

Then created some test data: (result from `http://localhost:1337/user`)

```
[
  {
    "photos": [
      {
        "owner": 1,
        "createdAt": "2015-07-08T17:52:37.854Z",
        "updatedAt": "2015-07-08T17:52:37.854Z",
        "id": 1
      },
      {
        "owner": 1,
        "createdAt": "2015-07-08T17:52:39.886Z",
        "updatedAt": "2015-07-08T17:52:39.886Z",
        "id": 2
      }
    ],
    "createdAt": "2015-07-08T17:52:18.501Z",
    "updatedAt": "2015-07-08T17:52:18.501Z",
    "id": 1
  },
  {
    "photos": [],
    "createdAt": "2015-07-08T17:52:21.212Z",
    "updatedAt": "2015-07-08T17:52:21.212Z",
    "id": 2
  }
]
```

And right now I'm trying to get all photos by specific user.

I followed documentation (http://sailsjs.org/documentation/reference/blueprint-api/populate-where)

and figured out that I need to GET on (in my example) : `http://localhost:1337/user/1/photos` but it returning `404` error.

I haven't changed anything in config directory.

Am I doing sth wrong? or it's a bug?

**Comment by giannoug** [comment]
I can confirm this issue. Here are the steps to reproduce.

Create a new Sails app

    $ sails new foo
    $ cd foo
    $ sails generate api purchase
    $ sails generate api employee

Edit `api/models/Purchase.js`

    module.exports = {

      attributes: {
      	     cashier: {
  		    model: 'Employee'
  	     }
      }
    };

Create some sample data:
`http://localhost:1337/employee/create/?name=%22Mr.%20T%22&age=25`
`http://localhost:1337/purchase/create/?name=%22Purchase%201337%22&cashier=1`

Now, navigate to `http://localhost:1337/purchase/1/cashier`. Server responds with 404 instead of the data.

**Comment by sailsbot** [comment]
Thanks for posting, @tbitowt.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by herrucules** [comment]
Hi, I'm also having this problem and got it sorted out.
I'm currently using version 0.11.2

*quick fix:*
at lib/hooks/blueprints/index.js 
go to line 376
**change**
`_bindRoute( _getAssocRoute('get %s/:parentid/%s/:id?'), 'populate', opts );
to
`_bindRoute( _getAssocRoute('get /%s/:parentid/%s/:id?'), 'populate', opts );
